### PR TITLE
docs: document metrics used for observability [RM-112]

### PR DESCRIPTION
**Pull request by kkunapuli**
## Ticket

[RM-112]

## Description

Documents metrics exposed by observability tooling, including pod labels. 

## Test Plan

None needed.

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

[RM-112]: https://hpe-aiatscale.atlassian.net/browse/RM-112?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-123]: https://hpe-aiatscale.atlassian.net/browse/DET-123?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tools/observability/README.md [hunk 1]
```diff
@@ -1 +1,35 @@
...
+| Key    | Value Description |
+| -------- | ------- |
+| determined.ai/container_id  |     |
+| determined.ai/experiment_id | `task_type=TRIAL` only    |
+| determined.ai/resource_pool | name of the resource pool, including `default`     |
+| determined.ai/task_id       |     |
+| determined.ai/task_type     | Determined task type, e.g. `TRIAL`, `NOTEBOOK`, `TENSORBOARD`     |
+| determined.ai/trial_id      | `task_type=TRIAL` only     |
+| determined.ai/user          | Determined username that initiated the request    |
+| determined.ai/workspace     | name of the workspace, including `Uncategorized    |
```
Comment (NicholasBlaskey) [inline tools/observability/README.md:31]:
```suggestion
| determined.ai/workspace     | name of the workspace, including `Uncategorized`    |
```

File: tools/observability/README.md [hunk 2]
```diff
@@ -1 +1,35 @@
-[See docs here](https://docs.determined.ai)
\ No newline at end of file
+[See docs here](https://docs.determined.ai)
+
+# Metrics
+Determined does not produce metrics, rather it uses existing tools to report information. Please see source documentation for details on specific metrics.
```
Comment (NicholasBlaskey) [inline tools/observability/README.md:4]: Did we ever decide on removing det state metrics? 

I know that seems to report a few metrics? Do you think we need to document those?

https://github.com/determined-ai/determined/blob/43289e9afcf780ec8b0bc7695b450408d391d2bf/master/internal/prom/det_state_metrics.go#L17

Comment (NicholasBlaskey) [inline tools/observability/README.md:4]: we should probably document the health metric https://github.com/determined-ai/determined/blob/43289e9afcf780ec8b0bc7695b450408d391d2bf/docs/integrations/prometheus/_index.rst?plain=1#L182

File: tools/observability/README.md [hunk 3]
```diff
@@ -1 +1,35 @@
-[See docs here](https://docs.determined.ai)
\ No newline at end of file
+[See docs here](https://docs.determined.ai)
+
+# Metrics
```
Comment (NicholasBlaskey) [inline tools/observability/README.md:3]: I know I said this was fine to put in the README but is there any reason why we shouldn't put it in the docs?

I think an advantage is the docs have the search functionality and everything will be in one place

**Review by NicholasBlaskey — APPROVED** [review]
A couple comments, but the docs look good to me

File: tools/observability/README.md (hunk 2 above)
Comment (kkunapuli) [reply tools/observability/README.md:4]: We didn't decide whether we were going to remove det state metrics or not ... but we did decide that we weren't going to document them or include them in any dashboards, etc going forward. For now, I think the philosophy is "don't stop customers from using them if they already are, but don't encourage anyone to start using them either".

File: tools/observability/README.md (hunk 3 above)
Comment (kkunapuli) [reply tools/observability/README.md:3]: I can put it in the docs; a lot of the links are to other github READMEs; putting it in the README felt like a similar level of formality but I understand what you're saying about searchability and discoverability.

File: tools/observability/README.md (hunk 2 above)
Comment (kkunapuli) [reply tools/observability/README.md:4]: Nope! This came up during planning; we are specifically not going to document the det state metrics to pave the away for future deprecation.

File: docs/integrations/observability/_index.rst [hunk 4]
```diff
@@ -194,3 +194,5 @@ Monitoring Dashboard
 
 -  Navigate to `127.0.0.1:9000 <http://127.0.0.1:9000>`__. Sign in with the ƒusername ``admin`` and
    the password you set above. You should see the ``Determined API Server Monitoring`` dashboard.
+
+For more information on reported metrics and how to use them, please see [determined/tools/observability/readme](https://github.com/determined-ai/determined/blob/main/tools/observability/README.md)
```
[the remark this answers is not in the document]
Comment (tara-det-ai) [reply docs/integrations/observability/_index.rst:198]: (remove "please"; replace "see" with "visit" - to match the style guide)

[the remark this answers is not in the document]
Comment (tara-det-ai) [reply docs/integrations/observability/_index.rst:198]: You'll also need a toctree which is required for every index page. something like this at the end of the file


.. toctree::
   :hidden:
   :glob:

   ./*/_index

File: tools/observability/README.md (hunk 3 above)
Comment (tara-det-ai) [reply tools/observability/README.md:3]: If you want the algolia search to index this content (and make it discoverable in search results), convert the README to a new page, maybe "metrics.rst"

File: docs/integrations/observability/_index.rst [hunk 5]
```diff
@@ -193,4 +193,80 @@ Monitoring Dashboard
...
+metrics with ``kube_pod_labels`` state metric. Some helpful resources (visit PromQL documentation for more details):
+
+-  `joining metrics <https://github.com/kubernetes/kube-state-metrics/tree/main/docs#join-metrics>`__ from kube-state-metrics 
+-  `matching vectors <https://prometheus.io/docs/prometheus/latest/querying/operators/#vector-matching>`__ from Prometheus
+
+.. toctree::
+   :hidden:
+   :glob:
+
+   ./*/_index
```
Comment (kkunapuli) [inline docs/integrations/observability/_index.rst:272]: @tara-det-ai `./*/_index` shows up in the rendered view ... I don't think that's supposed to happen?

File: docs/integrations/observability/_index.rst [hunk 6]
```diff
@@ -193,4 +193,80 @@ Monitoring Dashboard
...
 
 -  Navigate to `127.0.0.1:9000 <http://127.0.0.1:9000>`__. Sign in with the ƒusername ``admin`` and
-   the password you set above. You should see the ``Determined API Server Monitoring`` dashboard.
+   the password you set above. You should visit the ``Determined API Server Monitoring`` dashboard.
+
+******************
+ Metrics
+******************
+Determined does not produce metrics, rather it uses existing tools to report information. Visit 
+source documentation for details on specific metrics.
```
Comment (tara-det-ai) [inline docs/integrations/observability/_index.rst:202]: what is the URL for the source documentation that they should visit?

File: docs/integrations/observability/_index.rst [hunk 7]
```diff
@@ -193,4 +193,80 @@ Monitoring Dashboard
    kubectl -n det-monitoring port-forward svc/monitor-grafana 9000:80
 
 -  Navigate to `127.0.0.1:9000 <http://127.0.0.1:9000>`__. Sign in with the ƒusername ``admin`` and
-   the password you set above. You should see the ``Determined API Server Monitoring`` dashboard.
+   the password you set above. You should visit the ``Determined API Server Monitoring`` dashboard.
```
Comment (tara-det-ai) [inline docs/integrations/observability/_index.rst:196]: "fusername"?

-  Navigate to `127.0.0.1:9000 <http://127.0.0.1:9000>`__.
- Sign in with the username ``admin`` and your password.
- Navigate to the ``Determined API Server Monitoring`` dashboard.

File: docs/integrations/observability/_index.rst (hunk 5 above)
Comment (tara-det-ai) [reply docs/integrations/observability/_index.rst:272]: my fault, this page doesn't need a toctree, i was thinking there were more pages in the "observability" directory

File: docs/integrations/observability/_index.rst [hunk 8]
```diff
@@ -193,4 +193,80 @@ Monitoring Dashboard
...
+
+Kubernetes resource metrics and GPU metrics can be broken down by Determined resources by joining data 
+metrics with ``kube_pod_labels`` state metric. Some helpful resources (visit PromQL documentation for more details):
+
+-  `joining metrics <https://github.com/kubernetes/kube-state-metrics/tree/main/docs#join-metrics>`__ from kube-state-metrics 
+-  `matching vectors <https://prometheus.io/docs/prometheus/latest/querying/operators/#vector-matching>`__ from Prometheus
+
+.. toctree::
+   :hidden:
+   :glob:
```
Comment (tara-det-ai) [inline docs/integrations/observability/_index.rst:270]: toctree not needed afterall, my faul

File: docs/integrations/observability/_index.rst [hunk 9]
```diff
@@ -193,4 +193,80 @@ Monitoring Dashboard
...
+     - 
+   * - determined.ai/task_type
+     - Determined task type, e.g. ``TRIAL``, ``NOTEBOOK``, ``TENSORBOARD``
+   * - determined.ai/trial_id
+     - ``task_type=TRIAL`` only 
+   * - determined.ai/user
+     - Determined username that initiated the request 
+   * - determined.ai/workspace
+     - name of the workspace, including ``Uncategorized``
+
```
Comment (tara-det-ai) [inline docs/integrations/observability/_index.rst:261]: this table breaks the docs build

use a table editor like https://tableconvert.com/restructuredtext-generator

run `make fmt` then `make live`

File: docs/integrations/observability/_index.rst (hunk 6 above)
Comment (kkunapuli) [reply docs/integrations/observability/_index.rst:202]: The source documentation links are listed below. Would it make more sense to say "Visit source documentation listed below for details on specific metrics"?

Comment (tara-det-ai) [reply docs/integrations/observability/_index.rst:202]: ok makes sense. let's remove the sentence else they'll go looking for something that's already obvious enough.

******************
 Metrics
******************
Determined does not produce metrics, rather it uses existing tools to report information.

**Comment by tara-det-ai** [comment]
@kkunapuli i'll merge this while Circle CI is happy (else more conflicts are likely to creep in)

**Pull request merged**